osu.Game/Screens/Edit/Editor.cs

```diff
@@ -1137,6 +1138,62 @@ private void cancelExit()
             loader?.CancelPendingDifficultySwitch();
         }
 
+        public void SeekAndSelectHitObjects(string timestamp, Action<LocalisableString> onError)
+        {
+            string[] groups = EditorTimestampParser.GetRegexGroups(timestamp);
+
+            if (groups.Length != 2 || string.IsNullOrEmpty(groups[0]))
+            {
+                onError.Invoke(EditorStrings.FailedToProcessTimestamp);
+                return;
+            }
+
+            string timeGroup = groups[0];
+            string objectsGroup = groups[1];
+            string timeMinutes = timeGroup.Split(':').FirstOrDefault() ?? string.Empty;
+
+            // Currently, lazer chat highlights infinite-long editor links like `10000000000:00:000 (1)`
+            // Limit timestamp link length at 30000 min (50 hr) to avoid parsing issues
+            if (timeMinutes.Length > 5 || double.Parse(timeMinutes) > 30_000)
+            {
+                onError.Invoke(EditorStrings.TooLongTimestamp);
+                return;
+            }
+
+            double position = EditorTimestampParser.GetTotalMilliseconds(timeGroup);
+
+            editorBeatmap.SelectedHitObjects.Clear();
+
+            // Only seeking is necessary
+            if (string.IsNullOrEmpty(objectsGroup))
+            {
+                if (clock.IsRunning)
+                    clock.Stop();
+
+                clock.Seek(position);
+                return;
```

Comment:
To reduce code duplication, I'd seek first, early-return if the objects portion is empty, and handle selection at the very end.

Also, if we're unconditionally stopping clock, the seek should probably be smooth (`SeekSmoothlyTo()`)

Reply:
The reason for 2 Seek calls: whenever the objects portion is not empty, it moves to the next closest object instead of the given timestamp.

I can move the mentioned `nextObject` and `Seek` codes below position init to make them seem related and I'm not sure if we avoid code duplication as we would either need to:
- check `objectsGroup` twice - for position init and early return,
- or `Seek` twice - in early return and after setting the new position.

Sure we change `Seek()` to `SeekSmoothlyTo()`, though I like the snappy-Seek more when you click the link.